### PR TITLE
[SPARK-8877][MLlib]Public API for association rule generation

### PR DESCRIPTION
**Pull request by feynmanliang**
Adds FPGrowth.generateAssociationRules to public API for generating association rules after mining frequent itemsets.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ import org.apache.spark.rdd.RDD
  * association rules which have a single item as the consequent.
  */
 @Experimental
-class AssociationRules private (
+class AssociationRules private[fpm] (
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala]: Could still keep this private and use setter in FPGrowth.

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala [hunk 2]
```diff
@@ -40,7 +40,12 @@ import org.apache.spark.storage.StorageLevel
  * @tparam Item item type
  */
 @Experimental
-class FPGrowthModel[Item: ClassTag](val freqItemsets: RDD[FreqItemset[Item]]) extends Serializable
+class FPGrowthModel[Item: ClassTag](val freqItemsets: RDD[FreqItemset[Item]]) extends Serializable {
+  def generateAssociationRules(confidence: Double): RDD[AssociationRules.Rule[Item]] = {
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala]: Need API doc.

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala (hunk 1 above)
Comment (feynmanliang) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala]: I need the package private constructor to instantiate `AssociationRules` inside `FPGrowth#generateAssociationRules` (L44)

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala (hunk 2 above)
Comment (feynmanliang) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala]: OK

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala]: You can use the default constructor and then `setMinConfidence`.

Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala]: This is minor. You don't need to update the PR.

**Comment by mengxr** [comment]
LGTM. Merged into master. Please create JIRAs for the Python API and user guide. Thanks!

**Pull request closed without merging**